java/test/com/google/jglome/GlomeTest.java

```diff
@@ -0,0 +1,177 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.jglome;
+
+import static com.google.jglome.Glome.MAX_CNT_VALUE;
+import static com.google.jglome.Glome.MAX_TAG_LENGTH;
+import static com.google.jglome.Glome.MIN_CNT_VALUE;
+import static com.google.jglome.Glome.MIN_TAG_LENGTH;
+import static com.google.jglome.TestVector.TEST_VECTORS;
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.google.jglome.Glome.CounterOutOfBoundsException;
+import com.google.jglome.Glome.GlomeBuilder;
+import com.google.jglome.Glome.MinPeerTagLengthOutOfBoundsException;
+import com.google.jglome.Glome.WrongTagException;
+import java.security.InvalidKeyException;
+import java.util.Arrays;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Contains tests for com.google.jglome.Glome functionality.
+ */
+public class GlomeTest {
+
+  public static class KeyPair {
+
+    private byte[] publicKey;
+    private byte[] privateKey;
+
+    KeyPair(byte[] publicKey, byte[] privateKey) {
+      this.publicKey = publicKey;
+      this.privateKey = privateKey;
+    }
+
+    public byte[] getPrivateKey() {
+      return privateKey;
+    }
+
+    public byte[] getPublicKey() {
+      return publicKey;
+    }
+  }
+
+  final static int N_TEST_VECTORS = TEST_VECTORS.size();
+
+  Glome[][] glomeManagers = new Glome[N_TEST_VECTORS][2]; // first is for A, second is for B
+  KeyPair[] aKeys = new KeyPair[N_TEST_VECTORS];
+  KeyPair[] bKeys = new KeyPair[N_TEST_VECTORS];
+
+  GlomeTest() throws MinPeerTagLengthOutOfBoundsException, InvalidKeyException {
+    for (int i = 0; i < N_TEST_VECTORS; i++) {
+      TestVector testVector = TEST_VECTORS.get(i);
+      aKeys[i] = new KeyPair(testVector.getKa(), testVector.getKah());
+      bKeys[i] = new KeyPair(testVector.getKb(), testVector.getKbh());
+      glomeManagers[i][0] = new GlomeBuilder(bKeys[i].getPublicKey(), 32)
+          .setPrivateKey(aKeys[i].getPrivateKey())
+          .build();
+      glomeManagers[i][1] = new GlomeBuilder(aKeys[i].getPublicKey(), 28)
+          .setPrivateKey(bKeys[i].getPrivateKey())
+          .build();
+    }
+  }
+
+  @Test
+  public void testShouldFailWhenMinPeerTagLengthIsOutOfBounds() {
+    int[] minPeerTagLength = new int[]{MIN_TAG_LENGTH - 1, MAX_TAG_LENGTH + 1};
+
+    for (int len : minPeerTagLength) {
+      try {
+        new GlomeBuilder(aKeys[0].getPublicKey(), len);
+      } catch (MinPeerTagLengthOutOfBoundsException e) {
+        assertEquals(e.getMessage(),
+            String.format("minPeerTagLength argument should be in [%d..%d] range. Got %d.",
+                MIN_TAG_LENGTH, MAX_TAG_LENGTH, len));
+      }
```

Comment:
Do we need to do something if `new GlomeBuilder` does _not_ throw an exception?

Reply:
In this test the exception should be thrown. Added `assertThrows`.